webpack.config.js

```diff
@@ -173,21 +173,13 @@ const webpackConfig = {
 			getAliasesForExtensions()
 		),
 	},
-	node: {
-		console: false,
-		process: true,
-		global: true,
-		Buffer: true,
-		__filename: 'mock',
-		__dirname: 'mock',
-		fs: 'empty',
-		crypto: false,
-		stream: false,
-	},
+	node: false,
 	plugins: _.compact( [
+		new webpack.NormalModuleReplacementPlugin( /^fs$/, 'lib/mock' ),
 		new webpack.DefinePlugin( {
-			'process.env.NODE_ENV': JSON.stringify( bundleEnv ),
+			process: { env: { NODE_ENV: JSON.stringify( bundleEnv ) } },
```

Comment:
Just a not that `webpack` advises against this change

https://webpack.js.org/plugins/define-plugin/#feature-flags

> [this way it has changed] will overwrite the process object which can break compatibility with some modules that expect other values on the process object to be defined.

Reply:
not also that this directly undoes the cleanup from https://github.com/Automattic/wp-calypso/pull/15691

Reply:
Initially I was going to reply that this is intentional but clearly not ideal. It makes sense to only overwrite a specific property as a constant when polyfilling `process`, but what about when all of process should be `undefined` because we've disabled the polyfill?

I'm revisiting and trying to make it so that none of our code depends on _any_ non-explicitly included polyfills...which I didn't realize is harder than I thought.  `node: false` doesn't seem to be working like I thought it would!

Things like `import path from 'path'` and `import qs from 'querystring'` still work!

Reply:
I've re-cleaned it up. Thank you for pointing this out